### PR TITLE
Add graspologic-native

### PR DESCRIPTION
**Pull request by johandahlberg**
This PR adds a conda recipe for `graspologic-native` (https://github.com/microsoft/graspologic-native), the rust library companion of the python library `graspologic`. The python library will be added in a new PR once this has been merged.

Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [ ] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by johandahlberg** [comment]
@fbdtemme and @bdpedigo are you ok with being listed as maintainers for this package?

**Comment by fbdtemme** [comment]
Of course 🙂

**Comment by bdpedigo** [comment]
> @fbdtemme and @bdpedigo are you ok with being listed as maintainers for this package?

sure

**Comment by johandahlberg** [comment]
@conda-forge/help-python, ready for review!

**Comment by johandahlberg** [comment]
@conda-forge/help-python could some please have a look at this?

File: recipes/graspologic-native/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+build:
+  script:
+    - cargo bundle-licenses --format yaml --output THIRDPARTY.yml
+    - {{ PYTHON }} -m pip install . -vv --no-deps --no-build-isolation
+  number: 0
+
+requirements:
+  build:
+    - cargo-bundle-licenses
+    - python >=3.6,<3.12                     # [build_platform != target_platform]
```
Comment (xhochy) [inline recipes/graspologic-native/meta.yaml:21]: This should not be pinned.
```suggestion
    - python                     # [build_platform != target_platform]
```

**Review by xhochy — CHANGES_REQUESTED** [review]

**Comment by johandahlberg** [comment]
@ambarrio please verify that you want to be a maintainer of this package.

**Comment by ambarrio** [comment]
> @ambarrio please verify that you want to be a maintainer of this package.

Yes, please. Thank you!

**Comment by johandahlberg** [comment]
@xhochy thank you for the review! I have included your proposed change. Anything else you would like to see, or is this ready to merge?

**Comment by marcelm** [comment]
@conda-forge-admin, please ping @conda-forge/help-python this is ready to be merged 

(Not a maintainer or author, just interested in seeing this merged as well.)

File: recipes/graspologic-native/conda_build_config.yaml [hunk 2]
```diff
@@ -0,0 +1,2 @@
+MACOSX_DEPLOYMENT_TARGET: # [osx]
+  - 11.0 # [osx]
```
Comment (xhochy) [inline recipes/graspologic-native/conda_build_config.yaml:2]:
```suggestion
MACOSX_DEPLOYMENT_TARGET: # [osx and x86_64]
  - 11.0 # [osx and x86_64]
```

File: recipes/graspologic-native/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+    - cross-python_{{ target_platform }}     # [build_platform != target_platform]
+    - maturin >=1.4,<2.0                     # [build_platform != target_platform]
+    - {{ compiler("c") }}
+    - {{ compiler("rust") }}
+  host:
+    - pip
+    - python
+    - maturin >=1.4,<2.0
+  run:
+    - python
```
Comment (xhochy) [inline recipes/graspologic-native/meta.yaml:31]: You also need to add a run dependency on __osx for the deployment target here, see https://conda-forge.org/news/2024/01/29/rust-175-osx-sdk/

**Review by xhochy — CHANGES_REQUESTED** [review]

**Comment by johandahlberg** [comment]
@xhochy Thank you so much for the comments. I have addressed now. Let me know if you see any other issues.

**Review by xhochy — APPROVED** [review]

**Pull request merged**